File: 백준/Silver/10828. 스택/스택.c
#include <stdio.h>
#include <string.h>
#define STACK_SIZE 10000

int stack[STACK_SIZE];
int top = -1;

int isFull() {
	if (top >= STACK_SIZE - 1) {
		return 1;
	}
	else {
		return 0;
	}
}
int isEmpty() {
	if (top == -1) {
		return 1;
	}
	else {
		return 0;
	}
}

void push(int data) {
	if (!isFull()) {
		top++;
		stack[top] = data;
	}
}

int pop() {
	if (!isEmpty()) {
		printf("%d\n",stack[top--]);
	}
	else {
		printf("-1\n");
	}
}

int main() {
	int N,n;
	char s[10];
	scanf("%d", &N);
	for (int i = 0; i < N; i++) {
		scanf("%s", &s);
		if (strcmp(s,"push")==0) {
			scanf("%d",&n);
			push(n);
		}
		else if (strcmp(s, "pop") == 0) {
			pop();
		}
		else if (strcmp(s, "size") == 0) {
			printf("%d\n",top+1);
		}
		else if (strcmp(s, "empty") == 0) {
			printf("%d\n", isEmpty());
		}
		else if (strcmp(s, "top") == 0) {
						
			if (isEmpty()){
				printf("-1\n");
			}
			else {
				printf("%d\n",stack[top]);
			}
		}
	}

	return 0;
}

File: 백준/Silver/10828. 스택/스택.js
const fs = require('fs');
const input = fs.readFileSync("/dev/stdin").toString().trim().split('\n');
const N = Number(input[0]);
const stack = [];
const result = [];
for(let i =1; i<=N; i++){
    let cmd = input[i].split(' ');
    if(cmd[0] === 'push'){
        stack.push(cmd[1]);
    }
    else if(cmd[0] === 'pop'){
        if(stack.length == 0){
            result.push(-1);
        }
        else{
            result.push(stack.pop());
        }
    }
    else if (cmd[0] === 'size'){
        result.push(stack.length);
    }
    else if (cmd[0] === 'empty'){
        if(stack.length == 0){
            result.push(1);
        }
        else{
            result.push(0);
        }
    }
    else if(cmd[0] ==='top'){
        if(stack.length == 0){
            result.push(-1);
        }
        else{
            result.push(stack[stack.length-1]);
        }
    }
}

console.log(result.join('\n'));
